pkg/deploy/generator/resources_rp.go

```diff
@@ -553,7 +548,25 @@ firewall-cmd --add-port=444/tcp --permanent
 firewall-cmd --add-port=445/tcp --permanent
 firewall-cmd --add-port=2222/tcp --permanent
 
-cat >/etc/td-agent-bit/td-agent-bit.conf <<'EOF'
+export AZURE_CLOUD_NAME=$AZURECLOUDNAME
+
+az login -i
+az account set -s "$SUBSCRIPTIONID"
```

Comment:
```suggestion
az login -i --allow-no-subscriptions
```

The VMSS runs in a RG which is bound to a subscription already.  And with the gateway VM's we found the managed identity doesn't have permissions to as broad a set of subscriptions as RP.  Suggest we don't require a sub and we don't set the subscription.  Only suggesting here as the ` --allow-no-subscriptions` was removed as a part of this work..

Reply:
Thanks, I was afraid that --allow-no-subscriptions might default to some tenant wide change lower in the script instead of doing something subscription wide. But it should be safe over here (and there are no az commands except acr login/docker pulls).